### ISSUE TITLE
Barbarian Assault Helper

### ISSUE THREAD

**Issue by rsbfox** [rsbmatt/OSBuddy#51]
- [ ] Timer on call change. ? ticks from start of match. 1500 ticks after.
- [ ] Block player from using incorrect combat style.
- [ ] Highlight correct food to use in inventory.
- [ ] Highlight correct egg to pickup on right click menu.
- [ ] Mark different monsters on minimap and in game for each role.
- [ ] Annoying overlay if you have not called yet, noob.
- [ ] Overlay stating how many penance healers/penance runners/penance fighters or penance rangers are left that wave depending on what role you are & hp of the penance per wave.
- [ ] Overlay showing which healers are poisoned
- [ ] Highlight the most efficient path for dropping food to trap when your role is set to defender.
- [ ] Switch left click of calling horn to always be correct option.
- [ ] Notification when someone calls for medic. Not sure what would be best here. Could be system or just screen flash.
- [ ] Need to check if timers are accurate.

**Comment by rsnjames** [comment]
Added a couple of suggestions to the original post.

I think an overlay showing how many more monsters you have to kill/heal/lure to traps would be a great QOL update for BA. 

Defender was always my least favorite role to be in, so if there's a way we could make that role less of a pain that would be another nice update.

**Comment by Markipedia** [comment]
These are some great ideas you guys got, nice job!

I do have some suggestions in addition to these and also some possible issues that I can see coming with the ones you made.

One thing I would personally really like to see, if possible, is making the style that you are supposed to call the left click option. This would make the life for a casual player a lot easier as they don't have to right click and search for the style they have to call. It also makes the life of competitive players like myself easier as you don't have to focus on calling the right style anymore.

In addition to the overlay showing how many penance there are each wave, having the hp of the penance per wave would be very nice to have next to it.

When a player is getting low health and uses the "medic" option on their penance horn, a very small notification shows up on the screen of the healer. However this notification is way too small and can easily be overseen so it would be nice to have a bigger and more noticeable notification.

One of the main concerns I had with the suggestions you guys already made was with the call change timer. Barbarian assault is one of the worst coded aspects of the game and has a lot of glitches. One of these glitches is with the duration of the calls. Each call cycle is supposed to last exactly 30 seconds, but there are many, hard to explain, glitches that often lead to the call changes getting delayed thus making the calls more than 30 seconds long. I don't know much about programming but my biggest concern with this would be that stuff like blocking players from using incorrect combat styles and highlighting correct food in inventory could get broken by the inconsistent call lengths. I could obviously be completely wrong but it's something to look out for.

If you guys have any questions about the defender role in particular feel free to ask. The penance runner mechanics are extremely weird and hard to explain but I'll try to help wherever possible.

**Comment by rsbfox** [comment]
Thanks a lot. That is longest post on issue tracker so far.

Added all of that. As for your concern about timer issue. Timer would just be overlay with countdown. When call changes interfaces change on screen correct?

Those should be changing pvars like pest control overlay. Osbuddy would be watching them to know what overlays to draw or attack styles to block. Like if no one called it would block all styles.
Once call is made interface on screen would change triggering osbuddy to unblock correct attack style if that makes sense. The call timer is just a warning.

**Comment by bateams** [comment]
A tick counter for rs3 would be cool. The defenders would benefit largely off of that in rs3 and the attackers as well. Basically a counter at the corner of the screen that goes from 0-9 repeat and begins as soon as each wave starts. I can't imagine this would take much effort to make. I'd even pay for something like this.
